### DISCUSSION TITLE
Handle both REST and GraphQL requests together

### DISCUSSION THREAD

**Discussion by f0und3r** [mswjs/msw#654] [Q&A] [answered]
Hello!

Can MSW handling rest and graphql requests together?

I tried, but have got 404 on /graphql requests, if i comment rest handlers - all ok.

Thanks.

**Comment by kettanaito** [comment, marked as answer]
Hi, @f0und3r. 

MSW supports simultaneous handling of both REST and GraphQL API. You'd most likely want to scope the GraphQL handlers to a specific endpoint to prevent resource address collisions with REST API:

```js
import { graphql } from 'msw'

// Create a scoped handler.
const myGraphQLApi = graphql.link('http://your.server/graphql')

// Handle request in setupWorker/setupServer.
myGraphQLApi.query('QueryName', resolver)
```

[Read more about `graphql.link`](https://mswjs.io/docs/api/graphql/link).

> I tried, but have got 404 on /graphql requests

I suggest trying that again and following [uncaught requests debugging guidelines](https://mswjs.io/docs/recipes/debugging-uncaught-requests) if you get 404. If that page doesn't help, please attach your setup here so other contributors could help.

**Reply by f0und3r** [reply]
Thanks, all works fine now.
